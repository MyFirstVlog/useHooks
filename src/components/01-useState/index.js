import React from 'react';
import { CounterApp } from './CounterApp';
import { CounterWithCustomHook } from './CounterWithCustomHook';


const counterApp = <CounterApp/>
const counterWithCustomHook = <CounterWithCustomHook/>


module.exports = {
    counterApp,
    counterWithCustomHook
}