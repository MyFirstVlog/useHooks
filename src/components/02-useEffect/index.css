body {
    padding: 70px;
}
button {
    margin-top: 10px;
}