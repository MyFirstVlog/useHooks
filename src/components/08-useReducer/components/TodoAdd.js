import React from 'react'
import { useForm } from '../../hooks/useForm'

export const TodoAdd = ({ handleAddTodo}) => {

    //se pone description porque es el nombre del input
    const [{description},handleInputChange, reset] = useForm({
        description: ''
    })    

    const handleSubmit = (e) => {
        e.preventDefault()

        if(description.trim().length <= 1){
            return
        }

        const newTodo = {
            id : new Date().getTime(),
            desc : description,
            done: false
        }

        handleAddTodo(newTodo)
        reset()

    }

    return (
        <>
            <h4>Agregar TODO</h4>
            <hr/>

            <form className="d-grid gap-2" onSubmit={handleSubmit}>
                <input
                    type='text'
                    name='description'
                    placeholder = '¿qué debo hacer?'
                    autoComplete = 'off'
                    value={description}
                    onChange={handleInputChange}
                />
                
                <button
                    type='submit'
                    className='btn btn-primary mb-2 btn-block'                        >
                        Agregar
                </button>
            </form>
        </>
    )
}
