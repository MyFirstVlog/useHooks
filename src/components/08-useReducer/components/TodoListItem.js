import React from 'react'

export const TodoListItem = ({each, index,handleDelete = ()=>{} , handleToggle = ()=>{}}) => {
    return (
        <li
            key={each.id}
            className='list-group-item'
        >
            <p className={`${each.done && 'complete'}`}
                onClick = {() => handleToggle(each.id)}
            >
                    {index + 1}.{each.desc}
            </p>
            <button 
                className='btn btn-danger'
                onClick={()=> handleDelete(each.id)}
            >
                    Borrar
            </button>
        </li> 
    )
}
