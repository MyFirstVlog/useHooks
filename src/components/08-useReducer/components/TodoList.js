import React from 'react'
import '../styles.css'
import { TodoListItem } from './TodoListItem'
export const TodoList = ({todos=[], handleDelete = ()=>{} , handleToggle = ()=>{}}) => {
    return (
        <ul className='list-group list-group-flush'>
            {
                todos.map((each, index) =>(
                    <TodoListItem 
                                  key={each.id} 
                                  each={each} 
                                  index={index}
                                  handleDelete={handleDelete}
                                  handleToggle={handleToggle}
                    />
                ))
                
            }
        </ul>
    )
}
