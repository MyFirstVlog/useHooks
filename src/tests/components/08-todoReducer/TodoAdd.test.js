import React from 'react';
import { shallow } from "enzyme"
import { TodoAdd } from '../../../components/08-useReducer/components/TodoAdd';

describe('pruebas en <TodoAA/>', () => {
    
    const handleAddTodo = jest.fn()
    const wrapper = shallow(<TodoAdd
                                handleAddTodo= {handleAddTodo}
                            />)

    test('debe mostrarse correctamente ', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('no deria llmar handleAddTodo ', () => {
        const formSubmit = wrapper.find('form').prop('onSubmit')

        formSubmit({preventDefault(){}})

        expect(handleAddTodo).toHaveBeenCalledTimes(0)
    })

    test('debe de llamar la funcion handleAddTodo ', () => {
        //con un argumento
        const value = 'nihao mundo'
        wrapper.find('input').simulate('change',{
            target:{
                value,
                name:'description'
            }
        })
        
        const formSubmit = wrapper.find('form').prop('onSubmit')

        formSubmit({preventDefault(){}})

        expect(handleAddTodo).toHaveBeenCalledTimes(1)
        expect(handleAddTodo).toHaveBeenCalledWith( expect.any(Object))
        expect( handleAddTodo).toHaveBeenCalledWith({
            id:expect.any(Number),
            desc: value,
            done:false
        })

        expect(wrapper.find('input').prop('value')).toBe('')  // se miran los efectos mas que la implmentacion
    })
    
    
    


})
