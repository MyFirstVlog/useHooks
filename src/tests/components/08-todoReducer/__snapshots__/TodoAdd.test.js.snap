// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <TodoAA/> debe mostrarse correctamente  1`] = `
<Fragment>
  <h4>
    Agregar TODO
  </h4>
  <hr />
  <form
    className="d-grid gap-2"
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      name="description"
      onChange={[Function]}
      placeholder="¿qué debo hacer?"
      type="text"
      value=""
    />
    <button
      className="btn btn-primary mb-2 btn-block"
      type="submit"
    >
      Agregar
    </button>
  </form>
</Fragment>
`;
